### PR TITLE
Makes BlendShape maintain weight after merge

### PR DESCRIPTION
**Pull request by kibalab**
I made the BlendShape's weights are preserved after merging

**I left the question through the most recent commit. Check it**

### CODE REVIEW

**Comment by kibalab** [comment]
I have read the CLA Document and I hereby sign the CLA

File: Packages/net.euan.lazyoptimiser/Editor/MergeMeshes.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public struct SkinnedMeshAnimationReference
         [MenuItem("Tools/Lazy Optimiser/Print Mergeable Meshes")]
         public static void PrintMergeableMeshes()
         {
-            ProcessAvatar(Selection.activeGameObject);
+            ProcessAvatar(Selection.activeGameObject); // KIBA : Why is doDestroy false in this case?
```
Comment (euan142) [inline Packages/net.euan.lazyoptimiser/Editor/MergeMeshes.cs:26]: When printing merge-able meshes we don't want to actually alter the main version, only debug print what would be changed

Comment (kibalab) [reply Packages/net.euan.lazyoptimiser/Editor/MergeMeshes.cs:26]: I checked!

**Review by kibalab — COMMENTED** [review]
I understood

File: Packages/net.euan.lazyoptimiser/Editor/MergeMeshes.cs [hunk 2]
```diff
@@ -103,7 +103,13 @@ public static void ProcessAvatar(GameObject avatarGameObject, bool doDestroy = f
                         Debug.Log($"Grouping meshes: {string.Join(", ", group.Select(smr => smr.name))}");
                         AdjustAnimations(descriptor, skinnedMeshes[0], linkedAnimations.Where(kvp => skinnedMeshes.Contains(kvp.Key)).ToDictionary(a => a.Key, b => b.Value));
                         MeshUtil.MergeSkinnedMeshes(skinnedMeshes);
+                        
                         MeshUtil.MergeSkinnedMeshes(new List<SkinnedMeshRenderer> { skinnedMeshes[0] }, true, true); // Euan: Hack to merge same materials
+
+                        Debug.LogError(skinnedMeshes[0].gameObject.name);
```
Comment (euan142) [inline Packages/net.euan.lazyoptimiser/Editor/MergeMeshes.cs:109]: Don't believe we should be logging this, at least as an error and with no context

**Review by euan142 — CHANGES_REQUESTED** [review]
Aside from the log message looks good

**Review by kibalab — COMMENTED** [review]
Used to validate the processing sequence of Skinned Mesh Render I don't need it anymore! Thank you.

**Review by euan142 — APPROVED** [review]

**Comment by euan142** [comment]
Looks good, thanks for the contribution!

**Pull request merged**